### ISSUE TITLE
Registry filters don't work correctly for relationships

### ISSUE THREAD

**Issue by caffeineinc** [silverstripe/silverstripe-registry#49]
So in testing i found a bit of a problem with this module. It seems if the search field is a relationship (any type) the only match available is a "partial match" @see https://github.com/silverstripe/silverstripe-registry/blob/master/src/RegistryPageController.php#L357

This hinders the lookup as it's doing a partial match (equivalent of a `WHERE <searchcoluumn> LIKE "%<val>%";`  This is obviously a problem, as when you're searching for an integer, it needs to be exact. 

I don't think validating on the input type would work either, because then you couldn't search a name field for numbers, which could be a valid use case. 

I believe the solution needs to check the instance for searchable terms that might be relations, and then switch the "type" based on this. 

```
CWP 2.2.2 / SilverStripe 4.3
"name": "silverstripe/registry",
"version": "2.1.0",
```

**Comment by NightJar** [comment]
@caffeineinc could you please include the versions you're looking at (SilverStripe, Registry, etc.)

**Comment by caffeineinc** [comment]
Oh yup, sorry. ;)

**Comment by caffeineinc** [comment]
@NightJar  I've fixed up the tests for you now :)

**Comment by NightJar** [comment]
> @NightJar I've fixed up the tests for you now :)

[Are you sure](https://travis-ci.org/silverstripe/silverstripe-registry/builds/511726324?utm_source=github_status&utm_medium=notification)? :S

**Comment by caffeineinc** [comment]
Sigh. environmental issues.

**Comment by NightJar** [comment]
Resolved in #50
